### PR TITLE
added chatroom services

### PR DESCRIPTION
**Pull request by mohitreddy03**
# Description

Please include the summary of the change as well as the issue that is fixed (if applicable). Also if any dependencies that are required for this change to work.

Fixes # (issue)

## Type of change

Please choose the relavant type of change.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update
- [ ] Other

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Also summarise the process for us to reproduce.

- [x] Test A: checked through my emulator.
- [ ] Test B: Firebase testing labs
- [ ] Test C: Test through flutter test code
- [ ] Test D: Other (Please specify)

**Test Configuration**:
* Firmware version:
* Hardware:
* Toolchain:
* SDK:

# Checklist:

- [x] My code follows the style guidelines of this application
- [ ] I have checked the code working on my emulator
- [ ] I have commented my code for better understanding
- [ ] These changes generate no new warnings
- [ ] These changes do not effect the existing functionalities,

### CODE REVIEW

**Review by hayat-tamboli — APPROVED** [review]

**Comment by hayat-tamboli** [comment]
image
 ### you need to fix these

File: lib/services/chatroomServices.dart [hunk 1]
```diff
@@ -0,0 +1,53 @@
+import 'package:Runbhumi/models/message.dart';
+import 'package:cloud_firestore/cloud_firestore.dart';
+
+class chatroomService {
```
Comment (hayat-tamboli) [inline lib/services/chatroomServices.dart:4]: change this to PascalCase
**ChatroomService**

File: lib/services/chatroomServices.dart [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+        .collection("users")
+        .where('username', isEqualTo: searchField)
+        .snapshots();
+  }
+
+  getAllusers() {
+    return FirebaseFirestore.instance.collection("users").snapshots();
+  }
+
+  Future<bool> addChatRoom(chatRoom, chatRoomId) {
```
Comment (hayat-tamboli) [inline lib/services/chatroomServices.dart:16]: you need to return some future<bool> value

File: lib/services/chatroomServices.dart [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+    return FirebaseFirestore.instance
+        .collection("DirectChats")
+        .doc(chatRoomId)
+        .collection("chats")
+        .orderBy('time')
+        .snapshots();
+  }
+
+  getUsersDirectChats(String itIsMyName) async {
+    return await FirebaseFirestore.instance
```
Comment (hayat-tamboli) [inline lib/services/chatroomServices.dart:48]: useless await applied

**Comment by sbangariGIT** [comment]
Will change these ASAP

**Pull request merged**